src/Common/src/System/Net/Security/NegotiateStreamPal.Windows.cs

```diff
@@ -109,17 +109,19 @@ internal static string QueryContextAuthenticationPackage(SafeDeleteContext secur
             var outSecurityBuffer = new SecurityBuffer(resultBlob, SecurityBufferType.SECBUFFER_TOKEN);
 
             Interop.SspiCli.ContextFlags outContextFlags = Interop.SspiCli.ContextFlags.Zero;
+            // There is only one SafeDeleteContext type on Windows which is SafeDeleteSslContext so this cast is safe.
```

Comment:
Is there more than one on Unix?

Reply:
1 on macOS, 1 on Windows, 2 on Unix (negotiate stream uses a different derived one `SafeDeleteNegoContext`) which is why I assume `SafeDeleteContext` is abstract. SslStream uses `SafeDeleteContext` on all platforms so they all benefit.